sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -222,6 +222,11 @@ internal class DiagnosticScopeValidatingAsyncEnumerable<T> : AsyncPageable<T>
             private readonly MethodInfo _methodInfo;
             private readonly bool _overridesGetAsyncEnumerator;
 
+            //for mocking
+            protected DiagnosticScopeValidatingAsyncEnumerable()
```

Comment:
Where do we mock this type?

Reply:
when we use this to wrap pageable objects returned by clients.  Those now get instrumented since the elements inside them get are clients that need to instrumented